src/main/scala/org/constellation/domain/redownload/MissingProposalFinder.scala

```diff
@@ -20,12 +20,12 @@ class MissingProposalFinder(
     * @return Map where the key is the Id of a peer and the value is a set of heights with missing proposals from the peer
     */
   def findMissingPeerProposals(
-    majorityRange: HeightRange,
+    lookupRange: HeightRange,
     proposals: PeersProposals,
     peersCache: PeersCache
   ): Map[Id, Set[Long]] = {
     val receivedProposals = proposals.mapValues(_.keySet)
-    val bounds = boundedRange(majorityRange)
+    val bounds = boundedRange(lookupRange)
 
     (peersCache - selfId)
       .mapValues(
```

Comment:
BTW I think that on line 33 we can actually filter out the peers that don't have a joining height. We don't consider peers without joining height when picking majority, so we don't want to fetch proposal from them either.

Reply:
Wow, I just checked and you're right. I wasn't aware of that condition. Fixing it 👍